### ISSUE TITLE
Write tests for APP

### ISSUE THREAD

**Issue by sudip-mondal-2002** [Portfolio-Shop/portfolioshop#47]
**Is your feature request related to a problem? Please describe.**
The front-end currently doesn't have any tests written.

**Describe the solution you'd like**
Use Jest or any other Unit testing library to write tests for the react frontend of the project.

**Comment by utkarsh4321** [comment]
@sudip-mondal-2002  hi I want to work on it. I never worked on the GitHub action thing but I can do it. if you help me a little bit. Can you assign it to me?

**Comment by sudip-mondal-2002** [comment]
Please work on writing the tests first, GitHub action is very easy part, I can help you with that.

**Comment by utkarsh4321** [comment]
Hii on configuring this project I m facing some issues. When installing node-gyp. So can you tell me how I will resolve this?

**Comment by sudip-mondal-2002** [comment]
write tests using jest only.

**Comment by utkarsh4321** [comment]
ok

**Comment by utkarsh4321** [comment]
@sudip-mondal-2002 Hii sorry for the late response. I m unable to set up this project on my laptop. Can you tell me ways to set up this project?

**Comment by sudip-mondal-2002** [comment]
Sure, what's the issue? Have you followed the contribution.md file?

**Comment by utkarsh4321** [comment]
yes, can I use docker to run this project? it will work. or not?

**Comment by sudip-mondal-2002** [comment]
You may use vscode remote containers.

**Comment by utkarsh4321** [comment]
ok i will try that and update you

**Comment by utkarsh4321** [comment]
@sudip-mondal-2002 can you suggest any docs regarding that vscode remote container

**Comment by sudip-mondal-2002** [comment]
https://code.visualstudio.com/docs/remote/containers

The container is already configured, no need to configure it again.

**Comment by utkarsh4321** [comment]
ok

**Comment by utkarsh4321** [comment]
Hii @sudip-mondal-2002 can you tell me what about local.setting.json that need to configure na so where i will get those url.

**Comment by sudip-mondal-2002** [comment]
check the setup.sh, if you run that file that'll create it

**Comment by utkarsh4321** [comment]
@sudip-mondal-2002 can I use react-testing-library with jest? To test react component?

**Comment by sudip-mondal-2002** [comment]
Yes sure, you may use react-testing-library

**Comment by utkarsh4321** [comment]
ok

**Comment by utkarsh4321** [comment]
![image](https://user-images.githubusercontent.com/30936607/196757440-59eae0a9-58f4-46a7-b093-e5b448264870.png)
Here I m only seeing these pages so I Will write test cases for these pages only or there are more pages present in this app

**Comment by utkarsh4321** [comment]
@sudip-mondal-2002  Just tell me if I need to write test cases for all the pages or if there are some specific components only for those you want test cases. Just tell me this?

**Comment by sudip-mondal-2002** [comment]
You haven't configured the project correctly, that's why you can't see the login button, and so you can't enter the dashboard. We need tests for the dashboard.
